### PR TITLE
add documentation

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: test-suite/src/test/resources/application-test.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+microstream:
+  storage:
+    main:
+      storage-directory: build/microstream
+      channel-count: 4
+    secondary:
+      channel-count: 4
+      channel-directory-prefix: 'channel_'
+      data-file-prefix: 'channel_'
+      data-file-suffix: '.dat'
```
Comment (timyates) [inline test-suite/src/test/resources/application-test.yml:10]: Adding this is causing the tests to fail...

I'm not sure what `secondary` was for, but to get the tests passing, we need to change the constructor of `CustomerRepositoryImpl` to

```
    CustomerRepositoryImpl(@Named("main") EmbeddedStorageManager embeddedStorageManager) { // <1>
```

In both the Java and Groovy test-suites

Does secondary have a special meaning here?  Or is it there for future work?

**Review by timyates — CHANGES_REQUESTED** [review]
Tests failing as there's now 2 `EmbeddedStorageManager`'s

File: test-suite/src/test/resources/application-test.yml (hunk 1 above)
Comment (sdelamo) [reply test-suite/src/test/resources/application-test.yml:10]: `secondary`will become the name qualifier.

Comment (timyates) [reply test-suite/src/test/resources/application-test.yml:10]: Cool, so we just need to add `Named` to the test classes that inject EmbeddedStorageManager (as above)?

**Review by timyates — APPROVED** [review]

**Pull request merged**